src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserSafeHandles.cs

```diff
@@ -144,10 +161,17 @@ internal sealed class SNIHandle : SafeHandle
             bool flushCache,
             bool fSync,
             bool fParallel,
+			#if NETFRAMEWORK
+            TransparentNetworkResolutionState transparentNetworkResolutionState,
+            int totalTimeout,
+			#endif
             SqlConnectionIPAddressPreference ipPreference,
             SQLDNSInfo cachedDNSInfo)
             : base(IntPtr.Zero, true)
         {
+#if NETFRAMEWORK
+            RuntimeHelpers.PrepareConstrainedRegions();
```

Comment:
I don't think this needs to be ifdeffed, or if it does we should make it so that it doesn't. Constrained region calls on netcore are nops.